main.tf

```diff
@@ -26,11 +26,25 @@ resource "github_repository_environment" "repo_cd_environment" {
     }
   }
   deployment_branch_policy {
-    protected_branches     = var.restrict_environment_branches
-    custom_branch_policies = !var.restrict_environment_branches
+    protected_branches     = false
+    custom_branch_policies = var.restrict_environment_branches
   }
 }
 
+resource "github_repository_environment_deployment_policy" "ci_env_branch_restriction" {
+  count          = var.restrict_environment_branches ? 1 : 0
+  repository     = var.repo
+  environment    = "${var.env_id}-ci"
```

Comment:
Maybe it would be better to reference the environment name from the resource above?

Reply:
updated